csp/tests/test_examples.py

```diff
@@ -0,0 +1,26 @@
+
+import importlib
+import os.path
+import pytest
+import sys
+
+EXAMPLES_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "examples"))
+
+# make examples importable without keeping in source tree
+sys.path.append(EXAMPLES_ROOT)
+
+def _get_module(folder, filename):
+    # Don't fail during test gathering
+    try:
+        return importlib.import_module(f"{folder}.{filename.replace('.py', '')}")
+    except BaseException:
+        return None
+
+def _get_modules_to_test(folder):
+    return [(file, _get_module(folder, file)) for file in os.listdir(os.path.join(EXAMPLES_ROOT, "1_basics")) if file.endswith(".py")]
+
+class TestExamples:
+    @pytest.mark.parametrize("filename,module", _get_modules_to_test("1_basics"))
+    def test_1_basics(self, filename, module):
```

Comment:
can we test the other examples too?

Reply:
We should test all the examples, but most of the examples need to be redone so I figured don't bother testing examples we're going to (very soon) change